packages/components/src/components/checkbox/checkbox.tsx

```diff
@@ -161,7 +161,27 @@ export class Checkbox {
   public formResetCallback(): void {
     this.internals.setValidity({});
     this.internals.setFormValue(undefined);
-    this.checked = false;
+    this.checked = false; // TODO: Reset to the initial 'checked' state. Directly using this.host.hasAttribute('checked') doesn't work in frameworks like React, as it may not reflect the actual state. Need a more reliable way to track and restore the original checked value.
```

Comment:
Maybe makes sense here to research the prop `defaultChecked`, how it works and if we could utilize it.

Reply:
updated as discussed